quinn-proto/src/connection/mod.rs

```diff
@@ -2949,6 +2949,7 @@ impl Connection {
         let mut sent = SentFrames::default();
         let space = &mut self.spaces[space_id];
         let is_0rtt = space_id == SpaceId::Data && space.crypto.is_none();
+        space.pending_acks.on_send_scheduled();
```

Comment:
If we only call this in `populate_packet()`, how does that help us sending ACKs even if the connection isn't otherwise sending?

Reply:
Good question; I got some wires crossed here. As the comment in the definition suggests, we should be counting the number of packets we've *received*, not sent. Revising.

Reply:
This particular commit improves the behavior of the peer which is regularly sending ACK-eliciting data, but not receiving any. We're counting the number of non-ACK-eliciting packets received since we last sent an ACK, which is (probably) proportional to the state the receiver is keeping in memory. If `populate_packet` doesn't get called, then we're not generating traffic that must be ACKed, so there's no problem. If it does, then we emit an occasional spontaneous ACK to keep that count down.

Thoughts on how we might document this better?

Reply:
I guess I find the `on_send_scheduled()` method name (and the docstring) kind of opaque -- it doesn't do much to clarify what it's actually doing. For methods that do a bunch of admin work, naming after trigger event seems okay, but I think here we could be more specific? `maybe_ack_non_eliciting()`?

Reply:
SGTM; renamed and expanded doc.